server/server.cc

```diff
@@ -103,21 +104,26 @@ void Session::do_write(const std::string resp) {
 
 Server::Server(boost::asio::io_service& io_service, int port, HandlerConfiguration* handler, ServerStatus* status)
   : acceptor_(io_service, tcp::endpoint(tcp::v4(), port)),
-  socket_(io_service){
-  do_accept(handler, status);
+    socket_(io_service),
+    tp(2){
+  if(threadsOccupied < threadsSupported){
+    tp.schedule(boost::bind(&Server::do_accept, this, handler, status));
+    //threadsOccupied++
```

Comment:
I'm unsure of how to keep track of threads ending, and how to tell the client that all threads are being used. Also, considering using std::thread::hardware_concurrency() to determine number of threads to create